compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1279,9 +1279,8 @@ pub struct TargetOptions {
     /// `argc` and `argv` values.
     pub main_needs_argc_argv: bool,
 
-    /// Flag indicating whether ELF TLS (e.g., #[thread_local]) is available for
-    /// this target.
-    pub has_elf_tls: bool,
+    /// Flag indicating whether #[thread_local] is available for this target.
+    pub has_thread_local: bool,
```

Comment:
Maybe call it `has_object_thread_local`? This indicates whether thread locals are supported by the used object file format, not whether thread locals in general are supported. `thread_local!{}` works even if this is false.

Reply:
`has_target_thread_local` also makes sense. This matches the `#[cfg(target_thread_local)]` attribute whose value it controls.

Reply:
> `has_target_thread_local` also makes sense.

This is already in the context of `Target` though and many other fields don't have this prefix in target spec but has one as the cfg variable.

Some other options: `has_static_thread_local` (as in “static initailizers”), `has_native_thread_local`…